### DISCUSSION TITLE
**How did you guys solve the *Blockchain Basics NFT Challenge 👀* ?**

### DISCUSSION THREAD

**Discussion by Steiner-254** [Cyfrin/foundry-full-course-cu#1819] [Q&A]
**How did you guys solve this *Blockchain Basics NFT Challenge 👀* ?**

>> [Blockchain Basics (Arb)](https://arbiscan.io/address/0x2e99A4CA9c85383ccB54769adB3837437e296479) and [Blockchain Basics (Sepolia)](https://sepolia.etherscan.io/address/0x25056312685339b49e1d1C5a0b72Ff9eff13AF77)

**Comment by EngrPips** [comment]
Hello @Steiner-254, So when you click on any of the links you provided above, it will take you to the respective blockchain scan of the blockchain. You can go to the contract tab on the page and then click on the `write contract` tab, and that would display to you a list of functions, of which the `solve challenge` function is one of them which you need to call with the required argument to solve the challenge.

<img width="1440" alt="Screenshot 2024-06-11 at 11 00 55" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/30a95b6e-cf5a-4609-991a-6aa903a5ec8b">

When you click on the `solve challenge` method, it will open up a field for you to input the necessary argument to call the function. Input those arguments and send the transaction. Boom, you have solved the challenge and will get the NFT associated with the challenge to your address.

<img width="1440" alt="Screenshot 2024-06-11 at 11 05 31" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/a498d912-1a6e-41b2-a400-d9671ccaea6d">

**Reply by fels21** [reply]
You also have to click the "Connect to WEB3" button.

**Reply by EngrPips** [reply]
Indeed.

**Reply by Steiner-254** [reply]
> Hello @Steiner-254, So when you click on any of the links you provided above, it will take you to the respective blockchain scan of the blockchain. You can go to the contract tab on the page and then click on the `write contract` tab, and that would display to you a list of functions, of which the `solve challenge` function is one of them which you need to call with the required argument to solve the challenge.
> 
> <img alt="Screenshot 2024-06-11 at 11 00 55" width="1440" src="https://private-user-images.githubusercontent.com/95760918/338536710-30a95b6e-cf5a-4609-991a-6aa903a5ec8b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMDkzMjcsIm5iZiI6MTcxODEwOTAyNywicGF0aCI6Ii85NTc2MDkxOC8zMzg1MzY3MTAtMzBhOTViNmUtY2Y1YS00NjA5LTk5MWEtNmFhOTAzYTVlYzhiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjExVDEyMzAyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg5M2Q1YTU0NjFmZGJmNDg2MTJiMzk2MTdiODkxYTdiM2M3NjUxMjI1NGRkMjc5Nzk4MmUyOTc5Mjc1YzVhNGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AXcGmiJ5wd0cy2SPf9xvwr9AjlZExwG4rtU66Xe6R2Y">
> When you click on the `solve challenge` method, it will open up a field for you to input the necessary argument to call the function. Input those arguments and send the transaction. Boom, you have solved the challenge and will get the NFT associated with the challenge to your address.
> 
> <img alt="Screenshot 2024-06-11 at 11 05 31" width="1440" src="https://private-user-images.githubusercontent.com/95760918/338538501-a498d912-1a6e-41b2-a400-d9671ccaea6d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMDkzMjcsIm5iZiI6MTcxODEwOTAyNywicGF0aCI6Ii85NTc2MDkxOC8zMzg1Mzg1MDEtYTQ5OGQ5MTItMWE2ZS00MWIyLWE0MDAtZDk2NzFjY2FlYTZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjExVDEyMzAyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY5NjhkYzY5MWUwOWZlNDBjNTQzN2Q2ZTU5YTczMTQxYzc5MDgyMGViYzIyMTI0ODU5YThkNmJjNmQ1ODFjM2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.iAo2MEntEHa5wjoylhy7k-iBrqIdtH1mrXAvMYnbqUc">

Thanks ..lemme try <3

**Reply by EngrPips** [reply]
Anytime, Friend. Let us know how it plays out, and please, how did you reply to my text the way you did.

<img width="901" alt="Screenshot 2024-06-11 at 14 59 09" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/72e4b3e0-8251-485a-a2a9-d5b0b526615e">

**Reply by Steiner-254** [reply]
> Anytime, Friend. Let us know how it plays out, and please, how did you reply to my text the way you did.
> 
> <img alt="Screenshot 2024-06-11 at 14 59 09" width="901" src="https://private-user-images.githubusercontent.com/95760918/338613783-72e4b3e0-8251-485a-a2a9-d5b0b526615e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNjYxMjMsIm5iZiI6MTcxODE2NTgyMywicGF0aCI6Ii85NTc2MDkxOC8zMzg2MTM3ODMtNzJlNGIzZTAtODI1MS00ODVhLWEyYTktZDViMGI1MjY2MTVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDA0MTcwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmNzIwZTg3NzU0YWMwM2NlNmVkNjk1MzVlOWM3OTE2OTg3Mzk4ODRhN2FiMjFlNjdlZWQxNjc1YjFmMjYzYzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qfA9ygVrrmvdjy562lv273_OwlouDdv3ZnsRzrEaSqQ">

This way:
![test](https://github.com/Cyfrin/foundry-full-course-f23/assets/70921705/cdc7440e-4264-40b0-85f5-bc920d1c7a0e)

**Reply by EngrPips** [reply]
> > Anytime, Friend. Let us know how it plays out, and please, how did you reply to my text the way you did.
> > <img alt="Screenshot 2024-06-11 at 14 59 09" width="901" src="https://private-user-images.githubusercontent.com/95760918/338613783-72e4b3e0-8251-485a-a2a9-d5b0b526615e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNjYxMjMsIm5iZiI6MTcxODE2NTgyMywicGF0aCI6Ii85NTc2MDkxOC8zMzg2MTM3ODMtNzJlNGIzZTAtODI1MS00ODVhLWEyYTktZDViMGI1MjY2MTVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDA0MTcwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmNzIwZTg3NzU0YWMwM2NlNmVkNjk1MzVlOWM3OTE2OTg3Mzk4ODRhN2FiMjFlNjdlZWQxNjc1YjFmMjYzYzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qfA9ygVrrmvdjy562lv273_OwlouDdv3ZnsRzrEaSqQ">
> 
> This way: ![test](https://private-user-images.githubusercontent.com/70921705/338824012-cdc7440e-4264-40b0-85f5-bc920d1c7a0e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxNzYzNTQsIm5iZiI6MTcxODE3NjA1NCwicGF0aCI6Ii83MDkyMTcwNS8zMzg4MjQwMTItY2RjNzQ0MGUtNDI2NC00MGIwLTg1ZjUtYmM5MjBkMWM3YTBlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEyVDA3MDczNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM0Zjg1NDI5OTAzMTQ3YzQxZmUyMTlmNmYwODE0YjI0ODMwNzljMWU3NjA4ZWY4YjZhOWQ2YzYxZTk5NjhhNzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.68JEa9FgbscIK-HsVODxUApSmHpzx3eJ-OY5849o0RM)

Thank you.

**Reply by Steiner-254** [reply]
> you need to call with the required argument to s

Hey buddy, What's the format for the required string?

**Reply by EngrPips** [reply]
Just a string like this => "your data"

**Reply by saif1163** [reply]
hi @EngrPips  I m on foundry fundamental section 1 lesson 9  create a new foundry project ,i saw partick copying a code from Remix simple storage in github , and paste in src file in the foundry project , but i saw many exersizes and todo task in the video-------


![Screenshot (8)](https://github.com/user-attachments/assets/f999afe4-ea53-4899-88e5-76afb32ece65)
  

 when i go to copy that code from remix simple storage there is no file that contain a  code and there is no exersizes and todo task in github, does i miss something ? or i m searching wrong ? or are their removed from github and i m searching that code in README.md file.-----


![image](https://github.com/user-attachments/assets/275e41e6-4e59-4059-b62c-dcdbeefaffba)

**Reply by EngrPips** [reply]
The GitHub might have been updated, and things might have changed slightly.
